client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/pattern-large-preview.tsx

```diff
@@ -11,11 +12,53 @@ interface Props {
 	header: Pattern | null;
 	sections: Pattern[];
 	footer: Pattern | null;
+	activePosition: number;
 }
 
-const PatternLargePreview = ( { header, sections, footer }: Props ) => {
+// The pattern renderer element has 1px min height before the pattern is loaded
+const PATTERN_RENDERER_MIN_HEIGHT = 1;
+
+const PatternLargePreview = ( { header, sections, footer, activePosition }: Props ) => {
 	const translate = useTranslate();
 	const hasSelectedPattern = header || sections.length || footer;
+	const containerRef = useRef< HTMLUListElement | null >( null );
+
+	const renderPattern = ( key: string, pattern: Pattern ) => (
+		<li key={ key }>
+			<PatternRenderer patternId={ encodePatternId( pattern.id ) } />
+		</li>
+	);
+
+	useEffect( () => {
+		let timerId: number;
+		const scrollIntoView = () => {
+			const element = containerRef.current?.children[ activePosition ];
+			if ( ! element ) {
+				return;
+			}
+
+			const { height } = element.getBoundingClientRect();
+
+			// Use the height to determine whether the newly added pattern is loaded.
+			// If it's not loaded, try to delay the behavior of scrolling into view.
+			if ( height && height > PATTERN_RENDERER_MIN_HEIGHT ) {
```

Comment:
I think this may not the final height for the block, and it may lead to mismatch on scroll position. For example, when I add this block, its height will be changed after the scrolling

https://user-images.githubusercontent.com/10071857/210490963-aa72e966-f337-4269-8493-1308138bad4b.mp4

Reply:
It's expected. I don't think we need to wait until the pattern is loaded. What we need is to ensure the pattern has a height that is greater than the min-height so that we're able to scroll to the newly added pattern 🙂

Reply:
Okie